percona/controllers/pgc/pgc.go

```diff
@@ -0,0 +1,618 @@
+	}
+	if len(pgc.Spec.PGBouncer.AntiAffinityType) == 0 {
+		pgc.Spec.PGBouncer.AntiAffinityType = "preferred"
+	}
+	cluster.Spec.PodAntiAffinity = crv1.PodAntiAffinitySpec{
+		Default:    pgc.Spec.PGPrimary.AntiAffinityType,
+		PgBackRest: pgc.Spec.Backup.AntiAffinityType,
+		PgBouncer:  pgc.Spec.PGBouncer.AntiAffinityType,
+	}
+	cluster.Spec.Port = pgc.Spec.Port
+	cluster.Spec.Resources = pgc.Spec.PGPrimary.Resources.Requests
+	cluster.Spec.Limits = pgc.Spec.PGPrimary.Resources.Limits
+	cluster.Spec.User = pgc.Spec.User
+	cluster.Spec.UserLabels = pgc.Spec.UserLabels
+	cluster.Spec.SyncReplication = syncReplication
+	cluster.Spec.UserLabels = userLabels
+	cluster.Spec.Annotations.Global = specAnnotationsGlobal
+	cluster.Spec.Tolerations = pgc.Spec.PGPrimary.Tolerations
+	for _, s := range pgc.Spec.Backup.Storages {
+		switch s.Type {
+		case S3StorageType:
+			cluster.Spec.BackrestS3Bucket = s.Bucket
+			cluster.Spec.BackrestS3Endpoint = s.EndpointURL
+			cluster.Spec.BackrestS3Region = s.Region
+			cluster.Spec.BackrestS3URIStyle = s.URIStyle
+			cluster.Spec.BackrestS3VerifyTLS = strconv.FormatBool(s.VerifyTLS)
+		case GCSStorageType:
+			cluster.Spec.BackrestGCSBucket = s.Bucket
+			cluster.Spec.BackrestGCSEndpoint = s.EndpointURL
+			cluster.Spec.BackrestGCSKeyType = s.KeyType
+		}
+	}
+	cluster.Spec.BackrestRepoPath = pgc.Spec.Backup.RepoPath
+	cluster.Spec.BackrestStorageTypes = pgc.Spec.Backup.StorageTypes
+	cluster.Spec.PGDataSource.Namespace = pgc.Spec.PGDataSource.Namespace
+	cluster.Spec.PGDataSource.RestoreFrom = pgc.Spec.PGDataSource.RestoreFrom
+	cluster.Spec.PGDataSource.RestoreOpts = pgc.Spec.PGDataSource.RestoreOpts
+	cluster.Spec.ServiceType = pgc.Spec.PGPrimary.Expose.ServiceType
+	cluster.Spec.TLSOnly = pgc.Spec.TlSOnly
+	cluster.Spec.Standby = pgc.Spec.Standby
+	cluster.Spec.Shutdown = pgc.Spec.Pause
+	if cluster.Spec.TablespaceMounts == nil {
+		cluster.Spec.TablespaceMounts = make(map[string]crv1.PgStorageSpec)
+	}
+	for k, v := range pgc.Spec.TablespaceStorages {
+		cluster.Spec.TablespaceMounts[k] = v.VolumeSpec
+	}
+	cluster.Spec.WALStorage = pgc.Spec.WalStorage.VolumeSpec
+	cluster.Spec.PGDataSource = pgc.Spec.PGDataSource
+
+	return cluster
+}
+
+func getStorage(storageSpec *crv1.PgStorageSpec) crv1.PgStorageSpec {
+	newStorage := crv1.PgStorageSpec{
+		AccessMode:  "ReadWriteOnce",
+		Size:        "1G",
+		StorageType: "dynamic",
+	}
+	if storageSpec == nil {
+		return newStorage
+	}
+
+	if len(storageSpec.AccessMode) > 0 {
+		newStorage.AccessMode = storageSpec.AccessMode
+	}
+
+	if len(storageSpec.Size) > 0 {
+		newStorage.Size = storageSpec.Size
+	}
+
+	if len(storageSpec.StorageType) > 0 {
+		newStorage.StorageType = storageSpec.StorageType
+	}
+
+	newStorage.Name = storageSpec.Name
+	newStorage.StorageClass = storageSpec.StorageClass
+	newStorage.SupplementalGroups = storageSpec.SupplementalGroups
+	newStorage.MatchLabels = storageSpec.MatchLabels
+
+	return newStorage
```

Comment:
to not doing that much copying, what do you think about this one?
```suggestion
	if storageSpec == nil {
		return crv1.PgStorageSpec{
		AccessMode:  "ReadWriteOnce",
		Size:        "1G",
		StorageType: "dynamic",
	    }
	}

	if len(storageSpec.AccessMode) == 0 {
		storageSpec.AccessMode = "ReadWriteOnce"
	}

	if len(storageSpec.Size) == 0 {
		storageSpec.Size == "1G"
	}

	if len(storageSpec.StorageType) == 0 {
		storageSpec.StorageType = "dynamic"
	}
	
	return storageSpec
```